src/chainlit/server.py

```diff
@@ -306,11 +306,16 @@ async def serve_file(filename: str):
 @app.get("/{path:path}")
 async def serve(path: str):
     """Serve the UI."""
-    path_to_file = os.path.join(build_dir, path)
-    if path != "" and os.path.exists(path_to_file):
-        return FileResponse(path_to_file)
-    else:
-        return HTMLResponse(content=html_template, status_code=200)
+    app_file_path = os.path.join(config.root, path)
+    ui_file_path = os.path.join(build_dir, path)
+
+    file_paths = [app_file_path, ui_file_path]
+
+    for file_path in file_paths:
+        if file_path != "" and os.path.isfile(file_path):
```

Comment:
`file_path` can't be empty, right?
I would check whether `path` is empty before doing the `for` loop